src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -459,6 +461,25 @@ private EnvironmentOptions GetEnvironmentOptions(IProjectFile projectFile)
                 }
             }
 
+            try
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
+            catch (System.InvalidOperationException)
+            {
+                var msbuildExe = "";
+                var editions = new string[] { "Enterprise", "Professional", "Community" };
+                var targets = new string[] { "Microsoft.CSharp.targets", "Microsoft.CSharp.CurrentVersion.targets", "Microsoft.CSharp.CrossTargeting.targets" };
```

Comment:
I wasn't able to find a solid source online to verify this. I did do a cross-check with Mark about it. It would be helpful if I can get more suggestions on verifying this.

Reply:
I will do a quick check as well :)